bundles/org.pgcase.xobot.dbproc.jdbc/src/org/pgcase/xobot/dbproc/jdbc/functions/JdbcFunctionExtractor.java

```diff
@@ -0,0 +1,52 @@
+package org.pgcase.xobot.dbproc.jdbc.functions;
+
+import java.io.IOException;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Map;
+
+import org.pgcase.xobot.dbproc.jdbc.DbprocJdbc;
+import org.pgcase.xobot.dbproc.runtime.XIssueReporter;
+import org.pgcase.xobot.dbproc.runtime.functions.XFunctionDescriptor;
+import org.pgcase.xobot.dbproc.runtime.functions.XFunctionExtractor;
+
+public class JdbcFunctionExtractor implements XFunctionExtractor {
+	private static final String SCAN_DBPROC_FUNC_SQL_FILE = "/sql/dbproc_jdbc_functions.sql"; //$NON-NLS-1$
+	private static String sqlSentence = null;	
+
+	@Override
+	public Iterable<XFunctionDescriptor> extractFunctions(Object input, Map<String, Object> context, XIssueReporter reporter) {
+		if (input instanceof Connection) {
+			try (Connection jdbcConnection = (Connection) input) {		
+				ArrayList<XFunctionDescriptor> functions = new ArrayList<XFunctionDescriptor>();
+				String schema = context.getOrDefault("schema", (Object)"public").toString();
```

Comment:
String.valueOf(object) is safer than object.toString(). Also I would extract this to reusable utility like

public static String extractSchema(Map<String, Object> context)

Reply:
Extracted and moved. Done